apps/openmw/mwgui/settingswindow.cpp

```diff
@@ -272,10 +275,27 @@ namespace MWGui
         waterReflectionDetail = std::min(4, std::max(0, waterReflectionDetail));
         mWaterReflectionDetail->setIndexSelected(waterReflectionDetail);
 
+        int shadowsTextureSize = Settings::Manager::getInt ("shadow map resolution", "Shadows");
+        if (shadowsTextureSize >= 512)
+            mShadowTextureSize->setIndexSelected(0);
+        if (shadowsTextureSize >= 1024)
+            mShadowTextureSize->setIndexSelected(1);
+        if (shadowsTextureSize >= 2048)
+            mShadowTextureSize->setIndexSelected(2);
+        if (shadowsTextureSize >= 4096)
+            mShadowTextureSize->setIndexSelected(3);
```

Comment:
Not sure is it ok to call setIndexSelected multiple times in a row, but still this looks odd. You can swap the order and then it will be only one call.
```suggestion
        if (shadowsTextureSize >= 4096)
            mShadowTextureSize->setIndexSelected(3);
        else if (shadowsTextureSize >= 2048)
            mShadowTextureSize->setIndexSelected(2);
        else if (shadowsTextureSize >= 1024)
            mShadowTextureSize->setIndexSelected(1);
        else if (shadowsTextureSize >= 512)
            mShadowTextureSize->setIndexSelected(0);
```

Reply:
I just used a code for water RTT resolution as an example. Both should be fixed now.